### ISSUE TITLE
Tone down the Solar Hull

### ISSUE THREAD

**Issue by Vezzra** [freeorion/freeorion#1400]
As noted in two threads on our forum, here:

http://www.freeorion.org/forum/viewtopic.php?f=15&t=10450

...and here:

http://www.freeorion.org/forum/viewtopic.php?p=87784#p87784

...the Solar Hull needs some toning down, as it's really too overpowered. For 0.4.7 only some quick and easy tweaking is possible, so I suggest:

* Reduce the internal slots to 5
* Reduce the external slots to 15
* Reduce speed to 60
* Reduce min build time to 8
* Increase buils costs to 160

Or something along those lines. These numbers are just some rough ideas, but should give a general idea what direction I think we should go.

**Comment by Vezzra** [comment]
@freeorion/ai-team, please object if some of these changes would screw up the AIs ability to use the Solar Hull properly/effectively.

**Comment by afwbkbc** [comment]
I suggest leaving them as they are but making them unrepairable. If it's damaged - it will stay damaged. Free drydocking them is insane.

**Comment by geoffthemedio** [comment]
> Reduce min build time to 8

That doesn't sound like a nerf

**Comment by Vezzra** [comment]
> That doesn't sound like a nerf

The reason for that change isn't so much nerfing, but enabling a player to counter another player with Solar Hulls a bit more quickly. See the discussions I linked to above, a major problem is once you learn another player is going to field Solar Hulls against you, you are potentially doomed already.

In that situation every turn counts, so being able to field your own Solar Hulls two turns earlier might help. Not much, but at least a little bit. Together with slowing the Solar Hull down this is supposed to make the whole issue less bad. Not good, not even close, but just less bad. That has to suffice for the release.

**Comment by MatGB** [comment]
I did deliberately keep them longer to build than the others when I halved all the build times but am more than happy to see that reduced: another option I thought of was to unlock the solar hull building with an earlier tech (that's actually something I'd like to consider when we redo the tree, the buildings unlock earlier than the ships that use them).

I'm not too happy with reducing their slot totals too much, I am happy with increasing their costs, reducing their base structure and either reducing their speed a bit (they still need to be fast, they're the energy line flagship) or changing the way engines work/are costed.

I'm working through some options to test this week, but it does have to be done quickly: recosting the ships was something I had meant to do and completely forgot about, mea culpa.

**Comment by afwbkbc** [comment]
2 turns will make no difference, maybe 5 can.
Adding min build time reduction tech branch is better solution.

**Comment by MatGB** [comment]
>Adding min build time reduction tech branch is better solution.

Maybe, but it can't happen this cycle as would require AI work, we need to make changes for the forthcoming Release which need to be simple and require no AI work, we can work on bigger, longer term solutions later.

**Comment by Voker57** [comment]
Reducing speed of 100 should be enough. 60 would make it totally useless.

Also, how about buffing Sentient Hull / other two top level organic hulls to serve as counters? They are useless now. Giving +1 int/ext slots to both Bio-adaptive (currently slightly better and totally not worth its research version of Protoplasmic) and Ravenous (nowhere near the Quantum) and +2 ext + 1 int to Sentient should do it.

**Comment by afwbkbc** [comment]
Speed should not be lowered IMHO. Its energy flagship hull. It's the structure and cost that needs to be fixed. Other hull types should be buffed slightly aswell.
And long-term solution is making them unrepairable and adding min-build-time-reducing techs.

**Comment by MatGB** [comment]
> how about buffing Sentient Hull / other two top level organic hulls to serve as counters? They are useless now.

Already under consideration: I wouldn't say they're useless, but given the way stealth works they're a LOT less useful in multiplayer where you simply research detection and make them so—current plan is to do a lot with Stealth in the next cycle, and buffing/fixing the organic line a bit (not just for stealth) was something I hoped to fit in. That they are very low on external slots even on their best ships is a significant drawback and one I want to address.

Re the Solar and repair, I have considered in the past reducing the drydock repair and making an originating shipyard do better than drydock repairing, we dumped that idea overall but saying that the solar hull can only be repaired at a black hole with its shipyard would be an interesting and easily justifiable fix, but it can't go in before Release (unless there's a major delay, and even then that's likely to be AI related so can't give them even more work).

Also, for the record, you're preaching to the converted regarding Upkeep, it is possible right now to change the upkeep system if you guys want to, there's a macro for it and you can have it based on ship parts instead, far better and would make the solars a lot less affordable: we're also going to completely rework it at some point, hopefully, when we're ready to bring the Influence mechanic in.

**Comment by Voker57** [comment]
> Also, for the record, you're preaching to the converted regarding Upkeep, it is possible right now to change the upkeep system if you guys want to, there's a macro for it and you can have it based on ship parts instead, far better and would make the solars a lot less affordable: we're also going to completely rework it at some point, hopefully, when we're ready to bring the Influence mechanic in.

@MatGB Could you please give an example of such macro? I'm trying this but appear to be doing something wrong

```
'''(1 + 0.001 * ShipPartClassOwned  empire = Source.Owner Or [ class = ShortRange class = FighterHangar class = Armour class = Troops ] )'''
```

**Comment by Vezzra** [comment]
Re speed: I suggested the drastic speed cut because the extremely high speed of the Solar Hull was cited as one of the major issues (and because that makes sense for a massive ship like this). The problem simply is, with such a high base speed and so many internal slots, you can pump up a Solar Hull design to absolutely ridiculous speeds. Having such a strong hull also be far faster than anything else makes for the deadly mix we're dealing with here.

If you prefer to keep that speed (or only tone it down slightly), it will become very difficult to nerf the Solar Hull sufficiently, IMO. We would have to make it insanely expensive at least, and by that I mean build costs of 300 or somesuch.

> I'm not too happy with reducing their slot totals too much

Which is why I suggested 5 internal and 15 external slots. That shouldn't be too much IMO. The reduction of internal slots is a bit more aggressive, but that's to restrict the number of the engines you can slap on that hull, so the total speed can't get so ridiculously high anymore.

> I am happy with increasing their costs

Which needs to be a very big increase, especially if we want to keep the speed.

> reducing their base structure

Hm, the Solar Hull is supposed to be the most massive hull, so you'd expect it to have a very high structure. Of course, we can always say that because of it's nature it's structure is a bit unstable, or easy to destabilize, and account for that with a structure that would be "low" for it's supposed size.

What do you have in mind? 150, 100? I'd lean toward the latter, this would make it significantly more vulnerable.

> changing the way engines work/are costed

I'd rather not open that can of worms before the release.

> 2 turns will make no difference, maybe 5 can.

Although I basically agree, I'd also prefer to reduce that to 5, if we do that, this will totally throw off balance wrt to the min build times of all the other hulls. We'd have to revise that too, and this is becoming too much for 0.4.7. We need to stay with something where we only need to tweak the stats of the Solar Hull, not starting a cascade where we end up having to change a lot of other things too.

> Also, how about buffing Sentient Hull / other two top level organic hulls to serve as counters?

As @MatGB already said, already under consideration, but not for 0.4.7.

Please, guys, restict the suggestions here in the comment section of this issue to things we can reasonably do before the release. A lot of your suggestions are fine and interesting, but this issue requests a short-term fix for 0.4.7. Post suggestions that go beyond that in the forum threads I linked in the OP.

**Comment by LGM-Doyle** [comment]
My 2 cents.

Keep the high speed.  It is a signature of the line.  Use @Vezzra's suggestion and greatly reduce the structure.  Perhaps also couple this with an increase in the structure of the scattered asteroid hull.  Then each line has some advantage:
- solar        fast, long range
- asteroid   high structure and high shields
- organic    high stealth and high detection
- robotic     high repair and high supply(?)

**Comment by MatGB** [comment]
OK, starting work on this, appears that last time I did some work on them I'd already toned them down: they used to have 300 structure (ouch). I'd also reduced the structure of the other energy ships, I'm quite happy for them all to have relatively low structure given the nature of what they're supposed to be and fast/high slot are two big advantages anyway.

@afwbkbc note: I cannot do a lot of work on scripting unrepairable or reduced repair rates in time for the next formal release, that isn't going to happen. I know you guys play keeping up to date with the test releases but the formal release is an important step for us, complex scripting restrictions require careful balancing *and* work by the AI team that we don't have time for. I am happy to consider restrictions on repair post release, but I am very uncomfortable with the idea of a ship that's completely unrepairable.

Concentrate discussion on this issue to **simple stats adjustments** as that's all that's going to happen for 0.4.7 Release, after that we can discuss more complex changes but this issue isn't the place for scripting of unique attributes.

First pass: reduce build time to 8, unlock building with Fractal Hull tech, increase cost to 160, decrease structure to 80.

I suspect more might be needed for initial release but starting small and building up is my normal approach, I'll give it a game to see how it feels then put up a PR for Team Smac to give a go.

**Comment by Voker57** [comment]
Such a cost difference is nothing at that point of game. Less hull is right, but slots (in which you can stuff basically free armor and ndm) also must be reduced.

**Comment by afwbkbc** [comment]
MatGB: building should be unlocked by Fractal Hull tech AND Quantum Hull tech, otherwise player will be forced to always go with fractals.

"in which you can stuff basically free armor" - this is the problem of ridiculously low armor cost. Armor costs should be increased 3-4 times, and weapon costs decreased 2 times. Otherwise 100-20 ship is about 4 times more expensive than 20-100 ship, even though they are of equal power if going 1x1.
Can armor/weapon costs be fixed before next release? Armor is actually almost free now.

**Comment by afwbkbc** [comment]
Actually I meant - Fractal Hull OR Quantul Hull tech, whatever is discovered first.

**Comment by MatGB** [comment]
Actually I agree on that, I put it in for Quantum because I opened the wrong file but both makes most sense.

And while I've been meaning to do a balance on Parts, especially armour, for a long time, getting the numbers right would take multiple test iterations, so not time before release, but it is needed.

**Comment by afwbkbc** [comment]
All I can say is that armor is ridiculously cheap now and 1-weapon ship design (with armor in all other slots) seems to be the only viable design. If you add two weapons you will get squishier ship that would cost almost like two 1-weapon ships. So, 20/30 ship will cost just a tiny bit less than two 10/40 ships, and would obviously lose to them in fight. Weapon makes about 90% of total ship cost, so anything but adding almost free armor to one weapon does not work at all.

**Comment by geoffthemedio** [comment]
>@MatGB Could you please give an example of such macro? I'm trying this but appear to be doing something wrong

>```
>'''(1 + 0.001 * ShipPartClassOwned  empire = Source.Owner Or [ class = ShortRange class = FighterHangar class = Armour class = Troops ] )'''
>```

```
'''(1 + 0.001 * ShipPartsOwned empire = Source.Owner)'''
```
or
```
'''(1 + 0.001 * ShipPartsOwned empire = Source.Owner class = ShortRange)'''
```

As far as I know, the parser / function ShipPartsOwned doesn't support a list of classes, just one or no class specified. Using `Or` in that context won't work... the parser isn't that flexible, in part because the underlying `ValueRef` data structures aren't.

**Comment by Vezzra** [comment]
> First pass: reduce build time to 8, unlock building with Fractal Hull tech, increase cost to 160, decrease structure to 80.

> Such a cost difference is nothing at that point of game. Less hull is right, but slots (in which you can stuff basically free armor and ndm) also must be reduced.

Basically agree. Apparently there is a general consensus on keeping the speed, in this case a cost increase to 160 won't be sufficient. I'd say 200 *at least*, 250 or even 300 would be more like it IMO.

In turn I wouldn't decrease structure *that* much. 80 for a massive hull like this sounds ridiculous, keep it at 100 at least.

Also cut down the slots, at least a bit. 5 internal/15 external is still an *awful* lot, but at least not as bad as the current 8/18 (IIRC).

**Comment by Vezzra** [comment]
@MatGB, status report? Just asking for a heads up so I know how far along you are with this, and how long you think it's going to take you to finish it...

**Comment by Vezzra** [comment]
@MatGB... bump? Status on this...? Haven't seen much of you lately... I guess RL got you once again? 😉

Anyway, do you think you can finish this within the next few days? For the release we just need some minor adjustments, the numbers I proposed above should be fine without doing some extensive tests, worst case it's not enough and the Solar Hull still a bit overpowered (but at least not as much as before), or it's a bit too much and the Solar Hull ends up a bit overpriced (which also is far better than what we have now).

If you can't get to it anytime soon, someone else should take over. This should get into 0.4.7, but I don't want to delay the release for a fairly straightforward adjustment like this. If no one else gets around to do it, I will.

You can still continue to work on a more thorough revision after the release.

**Comment by MatGB** [comment]
Yeah, flu is almost gone now, I'm building the Solar Concentrator in my current test game right now. PArt of the problem is toning them down without making them then weaker than the Titan hull, which has 16/3/1 slots: I'm going to change the numbers/costs and comment out some slots, the biggest issue apart from the cost is the sheer speed they can get up, so post release I'll work on redoing the engine parts as I'm increasingly uncomfortable with the way you can stack them up on bigger ships and that I think is the main problem.

**Comment by Vezzra** [comment]
> the biggest issue apart from the cost is the sheer speed they can get up, so post release I'll work on redoing the engine parts as I'm increasingly uncomfortable with the way you can stack them up on bigger ships and that I think is the main problem.

I completely agree. Engine parts should not stack, that's definitely something I'd like to see fixed next release cycle.

**Comment by geoffthemedio** [comment]
> Engine parts should not stack

Or not exist at all, as was the original design?

**Comment by Vezzra** [comment]
Personally I don't have a strong preference for keeping or removing them. But with proper restrictions, why not keep them? If I remember the fluff explanations correctly, they aren't really *engine* parts, but something like secondary boosting systems which supposedly enhance the propulsion systems (thus making the ship faster).

So they actually just boost a certain stat of a hull (the speed), like other parts boost other stats (armor increases structure, stealth parts increase stealth). I see no reason why ship speed needs to be an exception here, unless of course we can't get the balance right.

**Comment by Voker57** [comment]
Stealth parts don't stack though, and armor competes with weaponry. Internal slots on non-carrier ships should have some meaningful alternative, or they become 'engine matrix slots' essentially.

**Comment by Vezzra** [comment]
Well, as more or less already agreed upon, the engine parts also should not stack, like shields and stealth parts, for similar reasons - these parts are/would be overpowered if they can stack.

Shields and stealth parts more so than engines of course, so it has been obvious that they can't stack, but as playtesting has revealed, the engine parts also are too powerful to be stackable.

> Internal slots on non-carrier ships should have some meaningful alternative

Yep, we need to come up with more parts for internal slots.

**Comment by MatGB** [comment]
Yup, damage control parts have been on the radar for ages, as have science and industry labs. I'd like to keep engines (given the core engine is my design I'm sorta biased ;-) ), but have them as just one optional refinement, in testing even reducing the Solar down to only just more slots than the Titan it's still really fast, and I've thought Titans can get too fast for ages as well.

The only reason I didn't make them non stacking ages ago was complete lack of other internals, now we have fighters that's not the case but we still need more options.

I think I'm happy with what I've got, need to complete my current test game and then I'll commit, I've only commented out the slots until we can do a proper pass with other stuff included once the engine parts are sorted: I think reworking the Interstellar Lighthouse/Logistics boost at the same time would make a lot of sense too.
